React i18n library ADR

mgurgel opened the pull request:
Resolves #357 and resolves #355

**Overall change:** This PR introduces a folder for collecting ADRs and adds two existing ones to it

**Code changes:**

- Two new text files in `/documentation/architecture-decision-records`
- _No code changes_

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval

mgurgel committed:
Started ADR folder

mgurgel committed:
Updated ADR001 and ADR002 with more info on the process

mgurgel committed:
Fixed heading order in ADR0001

dr3 commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,102 @@
+
+## Decision Outcome
+
+There was a clear preference among developers for option 1. `react-intl` as it is the most straight-forward and fully-supported option, providing the best developer experience out of all three. It was also the easiest to integrate into Psammead, as can be seen here: https://paper.dropbox.com/doc/2019-02-21-Component-Library-Meeting-i18n--AZgQeBdv~qXuQnL9jK5rcz4XAg-hSXxc23xnF6U7xnEkXwPc
```
Can we also write an option for rolling our own please? 

Devils advocate, writing a function to do
```
<FormattedMessage
    defaultMessage={`Hello {name}, you have {unreadCount, number}`}
    values={{name: <b>{name}</b>, unreadCount}}
/>
```
*Could* be not that tricky and might save us 10kb on the client for SPA

mgurgel replied:
So is this something that we're going to seriously consider and have meetings to discuss or is this just to include in the document? I would argue that if we do have the intention of rolling out our own i18n library, then we should take into account our looming deadline.

dr3 replied:
We should consider all possible options in an ADR, Theres a good chance someone can come to this document in 4 months time and go "why didnt they just make their own? would have been simple" How you consider it is up to you, but the document should say why you did/didnt go for it

mgurgel replied:
ADR amended 👍

mgurgel committed:
Added roll-your-own option to ADR0001

briankabiro commented on documentation/architecture-decision-records/ADR0002-type_groups.md:
```diff
@@ -0,0 +1,40 @@
+1. A theming library that applies different CSS styles depending on the service context
+[Link](https://github.com/bbc/psammead/issues/332)
+* Bad: adds another dependency to the stack
+* Bad: a stardadised solution may not take into account the particularities of Psammead
```
```suggestion
* Bad: a standardised solution may not take into account the particularities of Psammead
```

briankabiro commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+
+---
+
+1. `react-intl-native`
```
```suggestion
1. `react-intl-universal`
```

briankabiro commented on documentation/architecture-decision-records/ADR0002-type_groups.md:
```diff
@@ -0,0 +1,40 @@
+
+Technical Story:
+
+World Service has a UX requirement that GEL Type Styles (Canon, Long Primer, Double Pica, etc) should allow for mutable values of font size and line height across services. The possible permutations of such values were grouped into eight different 'scripts': arabic, burmese, cjk_thai, hindi_nepali_bengali, latin, mundo, sinhala, and tamil, as shown in the tabs of this spreadsheet https://docs.goo ...
```
```suggestion
World Service has a UX requirement that GEL Type Styles (Canon, Long Primer, Double Pica, etc) should allow for mutable values of font size and line height across services. The possible permutations of such values were grouped into eight different 'scripts': arabic, burmese, cjk_thai, hindi_nepali_bengali, latin, mundo, sinhala, and tamil, as shown in the tabs of [this spreadsheet](https://docs.google.com/spreadsheets/d/1WMY-rlSbekhB4pV3ojMuYoWDC71Qs93CB2wTfDwAsJg/edit#gid=1458344872)
```

briankabiro replied:
Turned this into an inline link.

briankabiro reviewed (commented):
Minor changes, otherwise the report looks good from my end.

amywalkerdev commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+
+---
+
+1. `react-intl-native`
```
```suggestion
1. `react-intl-universal`
```

Wrong library name

DenisBBC replied:
It should be 2. instead of 1.

briankabiro committed:
Update documentation/architecture-decision-records/ADR0001-i18n_library.md

Co-Authored-By: mgurgel <mgurgel@users.noreply.github.com>

briankabiro committed:
Update documentation/architecture-decision-records/ADR0002-type_groups.md

Co-Authored-By: mgurgel <mgurgel@users.noreply.github.com>

briankabiro committed:
Update documentation/architecture-decision-records/ADR0002-type_groups.md

Co-Authored-By: mgurgel <mgurgel@users.noreply.github.com>

amywalkerdev commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+
+## Context and Problem Statement
+
+Psammead components need a stardardised approach to localisation, including date and time formatting and string translation.
```
```suggestion
Psammead components need a standardised approach to localisation, including date and time formatting and string translation.
```

amywalkerdev commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+
+---
+
+1. `react-intl-universal`
```
```suggestion
2. `react-intl-universal`
```

amywalkerdev commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+
+Features:
+* It can do what we want i.e format numbers, times and dates
+* Can run in both the browser and Node js so we will be covered when we finally work on * ding SPA support )
```
```suggestion
* Can run in both the browser and Node js so we will be covered when we finally work on adding SPA support
```

amywalkerdev commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+* It can do what we want i.e format numbers, times and dates
+* Can run in both the browser and Node js so we will be covered when we finally work on * ding SPA support )
+* Supports fetching of config files from a remote source e.g S3
+* Seems like it was built for a specific reason; supporting translations in Vanilla JS. As * ’ll be using React components, I don’t think we have to worry about this.
```
```suggestion
* Seems like it was built for a specific reason; supporting translations in Vanilla JS. As we’ll be using React components, I don’t think we have to worry about this.
```

amywalkerdev reviewed (changes requested):
Just some spelling changes, everything else seems fine

DenisHernandez committed:
Fix spelling mistakes

DenisHernandez committed:
Remove * and replace it with the right text

DenisHdz commented:
Spelling mistakes corrected 👍🏻

amywalkerdev committed:
Merge branch 'latest' into adr

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+
+1. `react-intl`
+
+Created and maintained by yahoo.
```
```suggestion
Created and maintained by Yahoo.
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+# Adopt react-intl as i18n library
+## ADR-0001
+
+* Status: proposed
```
```suggestion
* Status: Proposed
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+
+This library provides React components and an API to format dates, numbers, and strings, including pluralization and handling translations.
+
+It uses and builds on the Internationalization API built-in to JavaScript.
```
```suggestion
It uses and builds on the [Internationalization API built-in to JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl).
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+* Display numbers with separators.
+* Display dates and times correctly.
+* Display dates relative to "now".
+* Pluralize labels in strings
```
```suggestion
* Pluralize labels in strings.
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+
+## Considered Options
+
+1. `react-intl`
```
```suggestion
1. [react-intl](https://github.com/yahoo/react-intl)
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+
+---
+
+2. `react-intl-universal`
```
```suggestion
2. [react-intl-universal](https://github.com/alibaba/react-intl-universal)
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+
+Created and maintained by Alibaba.
+
+It is an improvement of `react intl` in that it adds functionality for supporting translations in Vanilla JS. React-intl only does this in React Components.
```
```suggestion
It is an improvement of `react intl` in that it adds functionality for supporting translations in Vanilla JS. `react-intl` only does this in React Components.
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+
+Features:
+* It can do what we want i.e format numbers, times and dates
+* Can run in both the browser and Node js so we will be covered when we finally work on adding SPA support )
```
```suggestion
* Can run both in the browser and in NodeJS so we will be covered when we finally work on adding SPA support
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+* It can do what we want i.e format numbers, times and dates
+* Can run in both the browser and Node js so we will be covered when we finally work on adding SPA support )
+* Supports fetching of config files from a remote source e.g S3
+* Seems like it was built for a specific reason; supporting translations in Vanilla JS. As we’ll be using React components, I don’t think we have to worry about this.
```
> As we’ll be using React components, I don’t think we have to worry about this.

This sentence reads oddly - as if it's a consequence of the previous sentence. I think what you're getting at is this:

```suggestion
* Seems like it was built for a specific reason; supporting translations in Vanilla JS. This doesn't apply to us, as we’ll be using React components, but it doesn't prevent us from using it.
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+* Quick responses on both issues and Pull requests
+* 14 issues, 6 PRs
+
+* Developer experience:
```
```suggestion

Developer experience:
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+* Some parts of the docs aren’t grammatically correct. One will notice the errors but it’s not bad enough that one won’t be able to understand the docs.
+* Noticed that there are several issues that are asking for new features that are available on react-intl but not on react-intl-universal. Seems like they aren’t actively working on new features at the moment. If the current version suits our needs, we won’t have to worry about this.
+
+Licensing: BSD 3-Clause "New" or "Revised" License granted by Alibaba. License page info.
```
Where is the license page info link?

amywalkerdev replied:
Link added

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+
+---
+
+3. `fbt`
```
```suggestion
3. [fbt](https://github.com/facebookincubator/fbt)
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+
+3. `fbt`
+
+Created by facebook.
```
```suggestion
Created by Facebook.
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+
+Developer Experience:
+* Confusing syntax
+* Not many resources found, rather than the docs which are not very clear
```
```suggestion
* Not many resources found, other than the docs which are not very clear
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+* Confusing syntax
+* Not many resources found, rather than the docs which are not very clear
+* https://facebookincubator.github.io/fbt/docs/api_intro
+* https://twitter.com/fbopensource/status/1080883562293878784
```
Linking to this tweet doesn't feel very appropriate

amywalkerdev replied:
Removed

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+
+FBT works by transforming your `<fbt>` and `fbt(...)` constructs via Babel plugins. These plugins serve to extract strings from source and lookup translate payloads generated at build-time. FBT creates tables of all possible variations for the given fbt phrase and accesses this at runtime.
+
+Developer Experience:
```
```suggestion
Developer experience:

```

(Formatting for consistency)

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,109 @@
+
+## Decision Outcome
+
+There was a clear preference among developers for option 1. `react-intl` as it is the most straight-forward and fully-supported option, providing the best developer experience out of all three. It was also the easiest to integrate into Psammead, as can be seen here: https://paper.dropbox.com/doc/2019-02-21-Component-Library-Meeting-i18n--AZgQeBdv~qXuQnL9jK5rcz4XAg-hSXxc23xnF6U7xnEkXwPc
```
```suggestion
There was a clear preference among developers for option 1. `react-intl` as it is the most straight-forward and fully-supported option, providing the best developer experience out of all three. It was also the easiest to integrate into Psammead (according to our [component library meeting notes](https://paper.dropbox.com/doc/2019-02-21-Component-Library-Meeting-i18n--AZgQeBdv~qXuQnL9jK5rcz4XAg-hSXxc23xnF6U7xnEkXwPc)).
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0002-type_groups.md:
```diff
@@ -0,0 +1,40 @@
+# Components to use utility function to get CSS for type groups
+## ADR-0002
+
+* Status: proposed
```
```suggestion
Status: Proposed
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0002-type_groups.md:
```diff
@@ -0,0 +1,40 @@
+
+## Considered Options
+
+1. A theming library that applies different CSS styles depending on the service context
```
```suggestion
1. A theming library that applies different CSS styles depending on the service context. ([See our GitHub investigation](https://github.com/bbc/psammead/issues/332))
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0002-type_groups.md:
```diff
@@ -0,0 +1,40 @@
+## Considered Options
+
+1. A theming library that applies different CSS styles depending on the service context
+[Link](https://github.com/bbc/psammead/issues/332)
```
```suggestion
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0002-type_groups.md:
```diff
@@ -0,0 +1,40 @@
+* Bad: adds another dependency to the stack
+* Bad: a standardised solution may not take into account the particularities of Psammead
+
+
```
```suggestion
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0002-type_groups.md:
```diff
@@ -0,0 +1,40 @@
+* Bad: a standardised solution may not take into account the particularities of Psammead
+
+
+2. Utility functions specific to the Type Style that take the script as a parameter and return the relevant CSS [Link](https://github.com/bbc/psammead/issues/332#issuecomment-470638342)
```
```suggestion
2. Utility functions specific to the Type Style that take the script as a parameter and return the relevant CSS ([see comment on GitHub issue for more details](https://github.com/bbc/psammead/issues/332#issuecomment-470638342))
```

ChrisBAshton reviewed (changes requested):
Some formatting inconsistencies and helpful links. Otherwise good! 👍

ChrisBAshton committed:
Apply suggestions from code review

Co-Authored-By: amywalker1987 <44577498+amywalker1987@users.noreply.github.com>

Commit:
link changes

pjlee11 commented:
Is there any information on the size of the libraries? Just as a precursor to the decision, obviously in production we will attempt to tree-shake as much as possible.

Commit:
bundle sizes

Commit:
bundle sizes

amywalkerdev commented:
> Is there any information on the size of the libraries? Just as a precursor to the decision, obviously in production we will attempt to tree-shake as much as possible.


Bundlephobia links added for `react-intl` and `react-intl-universal`. `fbt` is not installable by npm so that was skipped, but it was the worst in terms of documentation & support anyway.

amywalkerdev committed:
Merge branch 'latest' into adr

dr3 commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,126 @@
+* Not many resources found, other than the docs which are not very clear
+* https://facebookincubator.github.io/fbt/docs/api_intro
+
+Support: 18 issues - 1 bug. Low activity in community
```
no Bundle Size?

briankabiro replied:
> Bundlephobia links added for react-intl and react-intl-universal. fbt is not installable by npm so that was skipped, but it was the worst in terms of documentation & support anyway.

^^

ChrisBAshton replied:
True: https://github.com/facebookincubator/fbt/issues/34

For completion we should try installing it as per the fbt README just so that the record says it adds xMB, but it sounds like the decision has already been made 👍

dr3 commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,126 @@
+Support: 18 issues - 1 bug. Low activity in community
+
+Licensing: FBT is MIT licensed, as found in the LICENSE file.
+
```
Needs a `---` line break

briankabiro replied:
```suggestion
---
```

dr3 commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,126 @@
+
+## Decision Outcome
+
+There was a clear preference among developers for option 1. `react-intl` as it is the most straight-forward and fully-supported option, providing the best developer experience out of all three. It was also the easiest to integrate into Psammead (according to our [component library meeting notes](https://paper.dropbox.com/doc/2019-02-21-Component-Library-Meeting-i18n--AZgQeBdv~qXuQnL9jK5rcz4XAg-hS ...
```
This document isnt public, can you please make it public or summarise the point about `the easiest to integrate` please

dr3 commented on documentation/architecture-decision-records/ADR0002-type_groups.md:
```diff
@@ -0,0 +1,38 @@
+* Mappings for each style must be easily editable and updateable
+* A component should be able to fetch the required CSS for a Type Style by invoking its name (Canon, Trafalgar) and its current script (arabic, mundo, etc)
+
+## Considered Options
```
Can you please talk about weight and tree shaking here

briankabiro replied:
Hey @dr3 . About tree shaking and weight, what exactly should be included? Is it how each package is handled based on the two?

dr3 replied:
The way that styles have been exported, such that they can be passed into a function, allows for the rest of the 'scripts' to not be pulled into the application. We should address this, saying why we did this, and how it avoids pulling them all in

briankabiro replied:
Ok, this is better. Going to try and write this up.

briankabiro replied:
I've made some changes and pushed them up. They are viewable [here](https://github.com/bbc/psammead/pull/364/commits/e76e60e9b848a3ed11536f6ddb4dbe0868f1c56a).

Comments welcome.

briankabiro committed:
Update documentation/architecture-decision-records/ADR0001-i18n_library.md

Co-Authored-By: amywalker1987 <44577498+amywalker1987@users.noreply.github.com>

Commit:
Merge branch 'latest' into adr

Commit:
Merge branch 'latest' into adr

briankabiro committed:
add example of using react-intl

- this updates the adr doc with an example of how to use the package with the react-intl package

briankabiro committed:
update type_groups ADR document

- this adds info to the document around how the utility functions solution affects application weight and tree-shaking

amywalkerdev committed:
Merge branch 'latest' into adr

dr3 commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,190 @@
+
+Support: 18 issues - 1 bug. Low activity in community
+
+Licensing: FBT is MIT licensed, as found in the LICENSE file.
```
Its currently a title due to being over a break

```suggestion
Licensing: FBT is MIT licensed, as found in the LICENSE file.

```

dr3 commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
````diff
@@ -0,0 +1,190 @@
+- ###### translations/en.json
+```javascript
+{
+  "headline": "This is a heading"
````
Headline is a bad example, I dont see any case in which a headline would use a translation

amywalkerdev replied:
There aren't many things in psammead currently which would need a translation like this. Is there a case where we might use the heading component outside of an article? It's mostly just a quick example to show how it works. We can amend it with better examples once we have other components.

dr3 reviewed (dismissed):
Just a couple things then LGTM

Commit:
Update documentation/architecture-decision-records/ADR0001-i18n_library.md

Co-Authored-By: amywalker1987 <44577498+amywalker1987@users.noreply.github.com>

amywalkerdev committed:
Merge branch 'latest' into adr

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,191 @@
+* Quick responses on both issues and Pull requests
+* 14 issues, 6 PRs
+
+
```
```suggestion
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,191 @@
+
+Size:
+
+Bundle Size
```
```suggestion
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,191 @@
+Size:
+
+Bundle Size
+75.4 kB Minified
```
```suggestion
- 75.4 kB Minified
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,191 @@
+
+Bundle Size
+75.4 kB Minified
+24.7 kB Minified + GZipped
```
```suggestion
- 24.7 kB Minified + GZipped
```

Won't display on separate lines in markdown unless we explicitly add `-` or `*` or `\n`

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,191 @@
+75.4 kB Minified
+24.7 kB Minified + GZipped
+
+https://bundlephobia.com/result?p=react-intl-universal@1.15.5
```
```suggestion
Source: [bundlephobia](https://bundlephobia.com/result?p=react-intl-universal@1.15.5)
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,191 @@
+
+https://bundlephobia.com/result?p=react-intl-universal@1.15.5
+
+Licensing: BSD 3-Clause "New" or "Revised" License granted by Alibaba. [License page info.](https://github.com/alibaba/react-intl-universal/blob/master/LICENSE.md)
```
```suggestion
Licensing: BSD 3-Clause "New" or "Revised" License granted by Alibaba. [License page info](https://github.com/alibaba/react-intl-universal/blob/master/LICENSE.md).
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,191 @@
+
+const locale = 
+  (navigator.languages && navigator.languages[0]) || 
+  navigator.langauge ||
```
```suggestion
  navigator.language ||
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,191 @@
+
+React Intl uses the provider pattern to scope an i18n context to a tree of components. This allows configuration like the current locale to be provided at the root of a component tree and made available to all the child components as well.
+
+The `Headline`(psammead) component wraps the `<FormattedMessage>` component of `react-intl` which is used for formatting string messages.
```
```suggestion
The `Headline` (Psammead) component wraps the `<FormattedMessage>` component of `react-intl` which is used for formatting string messages.
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
```diff
@@ -0,0 +1,191 @@
+
+The translations of our custom text messages will be stored for each language in a separate config file.
+
+- ###### translations/en.json
```
```suggestion
**translations/en.json**:
```

(The Accessibility Champion in me hates that we've gone from the markdown equivalent of `h2` to `h6` with no headings in between! 😁 )

ChrisBAshton commented on documentation/architecture-decision-records/ADR0001-i18n_library.md:
````diff
@@ -0,0 +1,191 @@
+}
+```
+
+- ###### translations/es.json
````
```suggestion
**translations/es.json**:
```

ChrisBAshton commented on documentation/architecture-decision-records/ADR0002-type_groups.md:
```diff
@@ -0,0 +1,62 @@
+
+Technical Story:
+
+World Service has a UX requirement that GEL Type Styles (Canon, Long Primer, Double Pica, etc) should allow for mutable values of font size and line height across services. The possible permutations of such values were grouped into eight different 'scripts': arabic, burmese, cjk_thai, hindi_nepali_bengali, latin, mundo, sinhala, and tamil, as shown in the tabs of [this spreadsheet](https://docs.g ...
```
```suggestion
World Service has a UX requirement that GEL Type Styles (Canon, Long Primer, Double Pica, etc) should allow for mutable values of font size and line height across services. The possible permutations of such values were grouped into eight different 'scripts': arabic, burmese, cjk_thai, hindi_nepali_bengali, latin, mundo, sinhala, and tamil, as shown in the tabs of [the spreadsheet "WS - Font Stacks & Styles Off Forge v1.0"](https://docs.google.com/spreadsheets/d/1WMY-rlSbekhB4pV3ojMuYoWDC71Qs93CB2wTfDwAsJg/edit#gid=1458344872)
```

Nice to know what we're about to click :)

ChrisBAshton commented on documentation/architecture-decision-records/ADR0002-type_groups.md:
````diff
@@ -0,0 +1,62 @@
+
+* We then pass the script into the utility typography functions depending on the particular font we want:
+```javascript
+getLongPrimer('arabic');
````
Do you pass the script or do you pass a string? Example above shows a string but based on the fact you're importing `{ arabic }` I'd expect it to actually be the object:

```suggestion
getLongPrimer(arabic);
```

ChrisBAshton reviewed (commented):
Some comments 👍

ChrisBAshton committed:
Apply suggestions from code review

Co-Authored-By: amywalker1987 <44577498+amywalker1987@users.noreply.github.com>

ChrisBAshton committed:
Update documentation/architecture-decision-records/ADR0002-type_groups.md

Co-Authored-By: amywalker1987 <44577498+amywalker1987@users.noreply.github.com>
